### PR TITLE
Redefine each field and add encoding info

### PR DESCRIPTION
**Pull request by bruth**
Signed-off-by: Byron Ruth b@devel.io

### CODE REVIEW

**Comment by bruth** [comment]
@l-k- I redefined the fields and reorganized the document. I hope the definitions are more clear.

File: 0.1.0.md [hunk 1]
```diff
@@ -1,63 +1,141 @@
...
+
+### `name`
+
+- Required: No
+- Type: string
+- Regexp: `^([a-z])([a-z][0-9]-._)*$`
+
+The name of the entity the [value](#value) is assigned to.
+
+The name is part of the *identity* of an entity, along with the [domain](#domain) if defined.
```
Comment (l-k-) [inline 0.1.0.md]: It's not 100% clear what "if defined" refers to -- name, domain, or both. 
Maybe something like this?
"If the name is defined, the [domain](#domain) must be defined as well. Together the name and domain define the _identity_ of an entity."

That's the only nitpick I had. I like this version a lot. The entity/value/state relationship is very clear, and version/revision/time is clear as well.

**Comment by bruth** [comment]
Thanks for reviewing. I pushed a change to clarify a few things. I updated the README description to better define the scope. While doing this a few (somewhat obvious) thoughts popped into my head:
- This is a format makes it possible to _do_ analytics on the data. Without a standard format with well-defined semantics, it would be difficult evaluate and track changes.
- This _is not_ intended to replace existing application-specific models/formats for database storage or API resource representations since these are optimized for their purposes.
- Without a set of tools that produce the [target analytics](https://github.com/chop-dbhi/factmsg/pull/1#issuecomment-149332007), the format is essentially useless.

The next steps will be to build out a few of these tools (I have prototypes for a few of them) and develop an application that standardizes the process of receiving and consuming the facts and display the analytics. The high-level components are:
- First line of storage for receiving facts as they come in. Kafka is a good choice for this due to it's durability, scalability, and pub/sub model.
- Suite of consumers that derive the analytics from the streams and store them.
- Set of databases or stores for storing the derived data. We could start with a single database such as Mongo to satisfy all of these requirements. As hotspots are detected we can migrate pieces of it to more optimized databases.
  - Object store - Store values.
  - Indexes -  Lookup entities by `domain`, `name`, `time`, `revision`, `version`, and `labels`
  - Graph - Store references between entities
  - Log - Store the events
  - Time series - Counts, etc.
- HTTP service to front all of resources and queries
- User interface for displaying the analytics, providing search, and display the log and values

**Comment by l-k-** [comment]
Saw your comment -- I will re-read and think about this later today.

**Comment by bruth** [comment]
Looking forward to it. I am writing up some more documentation to converge the original concepts of the Origins data model.

**Comment by bruth** [comment]
Updated the spec to include the `type` field for references. I also included a few types that we can discuss.

**Comment by bruth** [comment]
Thanks!

**Pull request merged**